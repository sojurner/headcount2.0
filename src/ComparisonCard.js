import React from "react";

export const ComparisonCard = ({ districtOne, districtTwo }) => {
  return <div />;
};
