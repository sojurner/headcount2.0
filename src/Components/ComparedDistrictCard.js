import React from "react";
import "../CSS/ComparedDistrictCard.css";
import PropTypes from "prop-types";

export const ComparedDistrictCard = ({ location, stats, index }) => {
  return (
    <div className={"compared-district-card" + index}>
      <h1 className="compared-location">{location}</h1>
      {Object.keys(stats).map((year, i) => {
        const avg = stats[year];
        return (
          <section
            key={`avgStats ${i}`}
            className={avg < 0.5 ? "year-red" : "year-black"}
            key={i}
          >
            {year}
            <span className={avg < 0.5 ? "year-red" : "year-black"}>
              ....
              {avg}
            </span>
          </section>
        );
      })}
      <p />
    </div>
  );
};

const { string, objectOf, func, number } = PropTypes;

ComparedDistrictCard.propTypes = {
  location: string,
  stats: objectOf(number),
  selectCard: func.isRequired
};
