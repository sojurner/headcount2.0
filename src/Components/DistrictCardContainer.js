import React from "react";
import PropTypes from "prop-types";
import { DistrictCard } from "./DistrictCard";
import "../CSS/DistrictCardContainer.css";

export const DistrictCardContainer = ({ districts, selectCard }) => {
  let counter = 0;

  const districtCard = districts.map((district, i) => {
    district.clicked ? counter++ : false;
    return (
      <DistrictCard
        key={i}
        {...district}
        selectCard={selectCard}
        counter={counter}
      />
    );
  });
  return <div className="district-card-container"> {districtCard} </div>;
};

const { shape, string, objectOf, number, func, arrayOf, bool } = PropTypes;

DistrictCardContainer.propTypes = {
  districts: arrayOf(
    shape({
      clicked: bool.isRequired,
      location: string,
      stats: objectOf(number)
    })
  ),
  selectCard: func.isRequired
};
