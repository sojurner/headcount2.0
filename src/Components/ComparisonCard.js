import React from "react";
import "../CSS/ComparisonCard.css";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div className="compare-data-card">
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] < comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyOne]}
        </p>
      </div>

      <p className="comparison-value">{comparisonData[comparisonKeys[2]]}</p>
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] > comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyTwo]}
        </p>
      </div>
    </div>
  );
};
