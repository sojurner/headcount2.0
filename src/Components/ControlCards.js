import React from "react";
import { ComparedDistrictCard } from "./ComparedDistrictCard";
import { ComparisonCard } from "./ComparisonCard";
import PropTypes from "prop-types";
import "../CSS/ControlCards.css";

export const ControlCards = ({
  comparisonData,
  selectCard,
  selectedDistricts
}) => {
  const controlCards = Object.keys(comparisonData).map((district, index) => {
    if (selectedDistricts.length === 2) {
      if (comparisonData[district].hasOwnProperty("location")) {
        return (
          <ComparedDistrictCard
            key={`compared ${index}`}
            selectCard={selectCard}
            location={district}
            stats={comparisonData[district].stats}
            index={index}
          />
        );
      }
      return (
        <ComparisonCard key={index} comparisonData={comparisonData[district]} />
      );
    }
  });
  return <div className="control-cards">{controlCards} </div>;
};

const { string, number, bool, shape, objectOf, func } = PropTypes;

ControlCards.propTypes = {
  comparisonData: objectOf(
    shape({
      location: string,
      stats: objectOf(number),
      clicked: bool
    })
    // comparisonData: objectOf(shape({
    //   compared: number
    // }))
  ),
  selectCard: func.isRequired
};
