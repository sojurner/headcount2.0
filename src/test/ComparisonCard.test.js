import React from "react";
import { ComparisonCard } from "../Components/ComparisonCard";

describe("ComparisonCard", () => {
  let shallowWrap;
  let mockDataOne = { "ACADEMY 20": 0.407, COLORADO: 0.53, compared: 0.768 };
  let mockDataTwo = { "ACADEMY 20": 0.7, COLORADO: 0.54, compared: 0.651 };

  beforeEach(() => {
    shallowWrap = shallow(<ComparisonCard comparisonData={mockDataOne} />);
  });

  it("should exist", () => {
    expect(shallowWrap).toBeDefined();
  });

  it("should match snapshot with no data", () => {
    shallowWrap = shallow(<ComparisonCard comparisonData={{}} />);
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot with first district avg greater than second district avg", () => {
    shallowWrap = shallow(<ComparisonCard comparisonData={mockDataTwo} />);
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot with second district avg greater than first district avg", () => {
    expect(shallowWrap).toMatchSnapshot();
  });
});
