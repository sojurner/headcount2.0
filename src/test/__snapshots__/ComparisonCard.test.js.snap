// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard should match snapshot with first district avg greater than second district avg 1`] = `
<div
  className="compare-data-card"
>
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-high"
    >
      0.7
    </p>
  </div>
  <p
    className="comparison-value"
  >
    0.651
  </p>
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-low"
    >
      0.54
    </p>
  </div>
</div>
`;

exports[`ComparisonCard should match snapshot with no data 1`] = `
<div
  className="compare-data-card"
>
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-high"
    />
  </div>
  <p
    className="comparison-value"
  />
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-high"
    />
  </div>
</div>
`;

exports[`ComparisonCard should match snapshot with second district avg greater than first district avg 1`] = `
<div
  className="compare-data-card"
>
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-low"
    >
      0.407
    </p>
  </div>
  <p
    className="comparison-value"
  >
    0.768
  </p>
  <div
    className="district-title"
  >
    Aggregate Avg.
    <p
      className="district-high"
    >
      0.53
    </p>
  </div>
</div>
`;
